jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridPresenter.java

```diff
@@ -162,70 +177,184 @@ public void getData( final Range visibleRange ) {
+
+    protected DataSetReadyCallback createDataSetDomainSpecificCallback(final int startRange, final int totalRowSize, final List<TaskSummary> instances){
+        return new DataSetReadyCallback() {
+            @Override
+            public void callback(DataSet dataSet) {
+                if(dataSet.getRowCount()>0) {
+                    Set<String> columns = new HashSet<String>();
+                    for (int i = 0; i < dataSet.getRowCount(); i++) {
+                        Long taskId = dataSetQueryHelperDomainSpecific.getColumnLongValue(dataSet, DataSetTasksListGridViewImpl.COLUMN_TASKID, i);
+                        String variableName = dataSetQueryHelperDomainSpecific.getColumnStringValue(dataSet,DataSetTasksListGridViewImpl.VARIABLE_NAME, i);
+                        String variableValue = dataSetQueryHelperDomainSpecific.getColumnStringValue(dataSet, DataSetTasksListGridViewImpl.VARIABLE_VALUE, i);
+
+                        for (TaskSummary task : instances) {
+                            if (task.getTaskId().equals(taskId)) {
+                                task.addDomainData(variableName, variableValue);
+                                columns.add(variableName);
+                            }
+                        }
+                    }
+                    view.addDomainSpecifColumns(view.getListGrid(), columns);
+                }
+                PageResponse<TaskSummary> taskSummaryPageResponse = new PageResponse<TaskSummary>();
+                taskSummaryPageResponse.setPageRowList(instances);
+                taskSummaryPageResponse.setStartRowIndex(startRange);
+                taskSummaryPageResponse.setTotalRowSize(totalRowSize);
+                taskSummaryPageResponse.setTotalRowSizeExact(true);
+                if (startRange + instances.size() == totalRowSize) {
+                    taskSummaryPageResponse.setLastPage(true);
+                } else {
+                    taskSummaryPageResponse.setLastPage(false);
+                }
+
+                updateDataOnCallback(taskSummaryPageResponse);
+            }
+
+            @Override
+            public void notFound() {
+                view.hideBusyIndicator();
+                errorPopup.showMessage("Not found DataSet with UUID [  variables ] ");
+                GWT.log("DataSet with UUID [  variables ] not found.");
+            }
+
+            @Override
+            public boolean onError(final ClientRuntimeError error) {
+                view.hideBusyIndicator();
+                errorPopup.showMessage("DataSet with UUID [  variables ] error: " + error.getThrowable());
+                GWT.log("DataSet with UUID [  variables ] error: ", error.getThrowable());
+                return false;
+            }
+        };
+    }
+
+    private TaskSummary createTaskSummaryFromDataSet(DataSet dataSet, int i) {
+        return new TaskSummary(
+                dataSetQueryHelper.getColumnLongValue( dataSet, DataSetTasksListGridViewImpl.COLUMN_TASKID, i ),
```

Comment:
Please use TaskDataSetConstants to consolidate all constants instead of DataSetTasksListGridViewImpl.